### PR TITLE
fix(Docker): transpile onbuild

### PR DESCRIPTION
**Pull request by cdaringe**
# problem statement

docker image un-runnable after `docker build .` cmd

# solution

patch.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,10 +11,9 @@ LABEL org.label-schema.name="greenkeeper-keeper" \
...
 ENV NODE_ENV=production
 
-RUN mkdir -p /app
-WORKDIR /app
-COPY . /app
+COPY . .
 EXPOSE 8080
-RUN npm install --prod
+RUN npm install --dev
+RUN npm run build
```
Comment (travi) [inline Dockerfile]: worth adding a `RUN npm prune --production` after the build to prune the weight of the dev dependencies after they're used for the `build` step?

Comment (cdaringe) [reply Dockerfile]: sure! comin in hot

Comment (cdaringe) [reply Dockerfile]: 41d608d

**Review by travi — APPROVED** [review]
This looks good. 

Once i merge it to get things working, I'm interested in your opinion on the alternative that comes to my mind. We could run the build step first and copy over the built files to the image rather than running the build on instance startup. Would trip startup time a bit, at least. Worth trying to go that route, or do you think this approach is good enough?

**Comment by cdaringe** [comment]
> We could run the build step first and copy over the built files to the image rather than running the build on instance startup

just to be clear, we are aligned that we don't build on container run, yes?  we build only on _image_ build.  there may be some optimizations.  the way its done now is definitely safest.  in ~1/2 my personal projects I build external of the ~container~ image-build-process and copy built assets in so i dont have to suffer thru the install times.  however, there can be risks w/ that too. `¯\_(ツ)_/¯`

**Comment by travi** [comment]
this is where me being rusty kicks in. sounds like youre saying that the `RUN` steps are part of image build process and that only the `CMD` step runs on container run? those details have gotten fuzzy for me. 

my intent was to avoid install and build at container run time, yes. so, if thats already the case with this approach, i'm fully on board. this is probably simpler to handle with the auto build process on docker hub too.

**Comment by travi** [comment]
did the build produced from merging your PR into mine end up working with those changes then (tagged `docker`)? if so, i'll get this merged into the mainline.

**Comment by cdaringe** [comment]
> so, if thats already the case with this approach, i'm fully on board

yep! thats the case

> did the build ... end up working

TBD.  did that get pushed to dockerhub?

when you're dev testing, what do you use to send a mock request to the service?

**Comment by travi** [comment]
yep theres a new build tagged as `docker` (based on branch name) [on docker hub](https://hub.docker.com/r/greenkeeperkeeper/reference-instance/builds/bergvk5sy8uessvsumghtua/)

**Comment by travi** [comment]
as far as testing goes, do you mean something more than how i use [`cucumber.js`](https://github.com/cucumber/cucumber-js) for [smoke testing](https://github.com/greenkeeper-keeper/reference-instance/tree/3fda7cfe6ffb9971c13f61620a6036fa50f7392a/test/integration/features) the reference instance? 

i use a [very similar approach](https://github.com/greenkeeper-keeper/hapi-greenkeeper-keeper/tree/master/test/integration/features) for integration testing the plugin itself, but those tests are far more involved.

is that the type of info youre looking for, or am i misunderstanding that question?

**Comment by cdaringe** [comment]
i wasnt very clear. "yes, kinda" to the smoke testing.  really i just want to fire some dummy payloads at the container, some from `gk` and some not from gk.  do you have a dummy payload example i can send?  ideally i could wire your integration tests to point at it, but its kind of hieroglyphics to me

**Comment by travi** [comment]
sorry for the delay on this. busy weekend. looks like you were able to test things out, but i'm happy to talk through more options if you think we should add something more

**Pull request merged**